apps/angular/1-projection/src/app/component/teacher-card/teacher-card.component.ts

```diff
@@ -1,40 +1,61 @@
-import { Component, OnInit } from '@angular/core';
-import { FakeHttpService } from '../../data-access/fake-http.service';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  OnInit,
+  inject,
+} from '@angular/core';
+import { first } from 'rxjs';
+import {
+  FakeHttpService,
+  randTeacher,
+} from '../../data-access/fake-http.service';
 import { TeacherStore } from '../../data-access/teacher.store';
-import { CardType } from '../../model/card.model';
-import { Teacher } from '../../model/teacher.model';
+import { CardItemDirective } from '../../ui/card/card-item.directive';
 import { CardComponent } from '../../ui/card/card.component';
+import { ListItemComponent } from '../../ui/list-item/list-item.component';
 
 @Component({
   selector: 'app-teacher-card',
   template: `
     <app-card
-      [list]="teachers"
-      [type]="cardType"
-      customClass="bg-light-red"></app-card>
+      [items]="teachers()"
+      customClass="bg-light-red"
+      (addItem)="addTeacher()">
+      <img image src="assets/img/teacher.png" width="200px" />
+      <ng-template cardItem let-teacher>
+        <app-list-item
+          (deleteItem)="deleteTeacher(teacher.id)"
+          [id]="teacher.id"
+          [name]="teacher.firstName" />
+      </ng-template>
+    </app-card>
   `,
-  styles: [
-    `
-      ::ng-deep .bg-light-red {
-        background-color: rgba(250, 0, 0, 0.1);
-      }
-    `,
-  ],
   standalone: true,
-  imports: [CardComponent],
+  imports: [
+    CardComponent,
+    ListItemComponent,
+    CardItemDirective,
+    CardComponent,
+    ListItemComponent,
+  ],
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class TeacherCardComponent implements OnInit {
-  teachers: Teacher[] = [];
-  cardType = CardType.TEACHER;
-
-  constructor(
-    private http: FakeHttpService,
-    private store: TeacherStore,
-  ) {}
+  readonly #store = inject(TeacherStore);
+  readonly teachers = this.#store.teachers;
+  readonly #http = inject(FakeHttpService);
 
   ngOnInit(): void {
-    this.http.fetchTeachers$.subscribe((t) => this.store.addAll(t));
+    this.#http.fetchTeachers$
+      .pipe(first())
```

Comment:
be careful with the `first` operator, because it will filter undefined and null result. You can have difficult bug to fix. You should prefer take(1) but even you don't have to use it here because http request complete on their own. You could use `takeUntilDestroyed` if you want to cancel the request if you leave your component

Reply:
Thank you, it's definitely valueable tip 😄